C@E Prefetch/prefetch_hls_fastly/src/main.rs

```diff
@@ -88,53 +88,61 @@ fn main(mut req: Request) -> Result<Response, Error> {
+                                format!("{}{}", base_path, segment.uri)
+                            );
                             //let segment_uri = base_url + segment.uri.;
-                            let pending_req_val = fastly::Request::head(format!("https://{}{}", base_path, segment.uri))
-                                .send_async(PRIMARY_BACKEND)?;
+                            let pending_req_val = fastly::Request::head(format!(
+                                "https://{}{}",
+                                base_path, segment.uri
+                            ))
+                            .send_async(PRIMARY_BACKEND)?;
                             pending_req_vec.push(pending_req_val);
                         }
                         for new_req in pending_req_vec {
                             new_req.wait()?;
                         }
                         //Ok(be_resp)
-                        let mut resp_vec= Vec::new();
-                        let resp_bytes = playlist_resp.write_to(&mut resp_vec).unwrap();
+                        let mut resp_vec = Vec::new();
+                        let _resp_bytes = playlist_resp.write_to(&mut resp_vec).unwrap();
                         // We respond with the new playlist
-                        Ok(Response::new()
-                            .with_body(Body::from(resp_vec)))
-
+                        Ok(Response::new().with_body(Body::from(resp_vec)))
                     }
-                    Err(_e) => Ok(be_resp)
+                    Err(_e) => Ok(be_resp),
                 }
             } else {
-                Err (fastly::Error::msg(be_resp.get_status().to_string()))
+                Err(fastly::Error::msg(be_resp.get_status().to_string()))
             }
         }
 
         (&Method::GET, url_path) if url_path.ends_with(".ts") => {
-
             let url = Url::parse(&req_clone.get_url().to_string())?;
-            let mut path_segments = url.path_segments().ok_or_else(|| "cannot be base").unwrap();
+            let path_segments = url.path_segments().ok_or_else(|| "cannot be base").unwrap();
             let path_segment_copy = path_segments.clone();
 
             // We save the originally requested file name since we will modify this request shortly.
             let file = path_segment_copy.last().unwrap().clone();
 
             let len = path_segments.clone().count();
-            let mut base_url = req_clone.get_url().host_str().unwrap();
+            let base_url = req_clone.get_url().host_str().unwrap();
             let mut path_url_vec = vec![];
             path_url_vec.push(base_url);
-            for (i, path) in path_segments.enumerate().filter(|&(index, path)| index < len-1) {
+            for (_i, path) in path_segments
+                .enumerate()
+                .filter(|&(index, _path)| index < len - 1)
+            {
                 path_url_vec.push(path);
-
             }
```

Comment:
if end up adding a one-liner code comment to the same block earlier in the code, maybe worth including here as well